src/treeffuser/samples.py

```diff
@@ -0,0 +1,207 @@
+from typing import List
+from typing import Literal
+from typing import Union
+
+import numpy as np
+from jaxtyping import Float
+from sklearn.neighbors import KernelDensity
+from tqdm import tqdm
+
+
+###################################################
+# Helper functions
+###################################################
+def _check_unidimensional(array) -> None:
+    if array.ndim > 2 and array.shape[-1] > 1:
+        raise ValueError("This method only applies to unidimensional responses.")
+
+
+###################################################
+# Main class
+###################################################
+class Samples:
+    def __init__(self, input_array: Float[np.ndarray, "n_samples batch y_dim"]):
```

Comment:
Do you want to add documentation to the class? to explain what's the point of it